spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java

```diff
@@ -437,6 +438,20 @@ private void prefetchMetadata(List<AppRegistration> appRegistrations) {
 		});
 	}
 
+	private void validateApplicationName(String name) {
+		char[] invalidWildCards = new char[]{':'};
+		StringBuilder invalidChars = new StringBuilder();
+		for (char invalidWildCard : invalidWildCards) {
+			if (name.contains(Character.toString(invalidWildCard))) {
+				invalidChars.append("'").append(invalidWildCard).append("'");
+			}
+		}
+
+		if (!invalidChars.toString().equals("")) {
+			throw new IllegalArgumentException("Application name: " + name + " cannot contain: " + invalidChars);
```

Comment:
It would be nice if we add single quotes around the `name` value. I will add it when merging.